### PR TITLE
usaco-1228 nlogn solution

### PR DESCRIPTION
**Pull request by envyaims**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

also module stuff idk

### CODE REVIEW

File: solutions/usaco-1228 [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+author: Chongtian Ma
+---
+
+[Official Analysis](http://www.usaco.org/current/data/sol_prob2_bronze_open22.html)
+
+## Solution 1
+
+See the official analysis.
+
+## Solution 2 - Sorting
```
Comment (SansPapyrus683) [inline solutions/usaco-1228:14]:
```suggestion
## Alternate Explanation
```

File: solutions/usaco-1228 [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+An important observation is that if Bessie is at position $x$, then the number of cows lying are all the
+cows with position less than $x$ that says "L" plus all the cows with a position greater than x that says
+"G". Thus, we can loop through every cow's position and treat it as point $x$ and compute the total number 
+of liars by using a forwards loop and a backwards loop. The answer is the minimum liars over all possible position $x$.
+
+## Implementation
+
+Time Complexity: $\mathcal{O}(NlogN)$
```
Comment (SansPapyrus683) [inline solutions/usaco-1228:23]:
```suggestion
**Time Complexity**: $\mathcal{O}(N \log N)$
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-1228 (hunk 1 above)
Comment (envyaims) [reply solutions/usaco-1228]: well its not really alternate because the official doesn't mention the observation lol.
its more like "additional" i guess?

**Review by SansPapyrus683 — APPROVED** [review]
in the future please put whitespace between your operators & control flow statements

File: solutions/usaco-1228 (hunk 1 above)
Comment (SansPapyrus683) [reply solutions/usaco-1228]: hm
but your algorithm is n log n while official sol is n^2
best use this

**Review by jessechoe10 — APPROVED** [review]

File: solutions/usaco-1228 [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+			/*
+			 * If the cow before says our position is to the left
+			 * but their position is strictly less than our position,
+			 * they're lying.
+			*/
+			lying_left[i]++;
+		}
+	}
+	
+	// Fill up lying_right in much the same way.
```
Comment (brad182) [inline solutions/usaco-1228:68]:
```suggestion
	// Fill up lying_right the same way.
```

File: solutions/usaco-1228 [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+	sort(cows.begin(), cows.end());
+	
+	/*
+	 * lying_left[i] stores the number of cows to the left of cow i
+	 * that must be lying given that Bessie is at the position of cow i.
+	 * 
+	 * lying_right stores the same thing, but does it so for the cows
+	 * to the *right* of i.
+	 */
+	vector<int> lying_left(n);
```
Comment (brad182) [inline solutions/usaco-1228:51]: i personally don't like _ in variable names, but that's probably optional

**Review by brad182 — APPROVED** [review]

File: solutions/usaco-1228 (hunk 4 above)
Comment (SansPapyrus683) [reply solutions/usaco-1228:51]: snake_case is standard for c++

File: solutions/usaco-1228 (hunk 3 above)
Comment (SansPapyrus683) [reply solutions/usaco-1228:68]: not really "the same way", there's some variations

**Pull request merged**